src/main/java/com/medallia/word2vec/Word2VecModel.java

```diff
@@ -76,69 +79,113 @@ public static Word2VecModel fromTextFile(File file) throws IOException {
 	}
 
 	/**
-   * Forwards to {@link #fromBinFile(File, ByteOrder)} with the default 
-   * ByteOrder.LITTLE_ENDIAN
-   */
-  public static Word2VecModel fromBinFile(File file)
-      throws IOException {
-    return fromBinFile(file, ByteOrder.LITTLE_ENDIAN);
-  }
-
-  /**
-   * @return {@link Word2VecModel} created from the binary representation output
-   * by the open source C version of word2vec using the given byte order.
-   */
-  public static Word2VecModel fromBinFile(File file, ByteOrder byteOrder)
-      throws IOException {
-
-    try (FileInputStream fis = new FileInputStream(file);) {
-      DataInput in = (byteOrder == ByteOrder.BIG_ENDIAN) ?
-          new DataInputStream(fis) : new SwappedDataInputStream(fis);
-
-      StringBuilder sb = new StringBuilder();
-      char c = (char) in.readByte();
-      while (c != '\n') {
-        sb.append(c);
-        c = (char) in.readByte();
-      }
-      String firstLine = sb.toString();
-      int index = firstLine.indexOf(' ');
-      Preconditions.checkState(index != -1,
-          "Expected a space in the first line of file '%s': '%s'",
-          file.getAbsolutePath(), firstLine);
-
-      int vocabSize = Integer.parseInt(firstLine.substring(0, index));
-      int layerSize = Integer.parseInt(firstLine.substring(index + 1));
-
-      List<String> vocabs = Lists.newArrayList();
-      List<Double> vectors = Lists.newArrayList();
-
-      for (int lineno = 0; lineno < vocabSize; lineno++) {
-        sb = new StringBuilder();
-        c = (char) in.readByte();
-        while (c != ' ') {
-          // ignore newlines in front of words (some binary files have newline,
-          // some don't)
-          if (c != '\n') {
-            sb.append(c);
-          }
-          c = (char) in.readByte();
-        }
-        vocabs.add(sb.toString());
-
-        for (int i = 0; i < layerSize; i++) {
-          vectors.add((double) in.readFloat());
-        }
-      }
-
-      return fromThrift(new Word2VecModelThrift()
-          .setLayerSize(layerSize)
-          .setVocab(vocabs)
-          .setVectors(vectors));
-    }
-  }
-
-  /**
+	 * Forwards to {@link #fromBinFile(File, ByteOrder)} with the default
+	 * ByteOrder.LITTLE_ENDIAN
+	 */
+	public static Word2VecModel fromBinFile(File file)
+			throws IOException {
+		return fromBinFile(file, ByteOrder.LITTLE_ENDIAN);
+	}
+
+	/**
+	 * @return {@link Word2VecModel} created from the binary representation output
+	 * by the open source C version of word2vec using the given byte order.
+	 */
+	public static Word2VecModel fromBinFile(File file, ByteOrder byteOrder)
```

Comment:
I switched the whole thing over to using `ProfilingTimer`. I don't quite get what it's for. Normal Java logging frameworks already let you choose whether you want to see the output or not, and `ProfilingTimer` has no log levels.